### PR TITLE
fix post install script with upstart

### PR DESCRIPTION
**Pull request by ipa-fmw**
- fixes issues left from https://github.com/ipa320/setup_cob4/pull/147 and https://github.com/ipa320/setup_cob4/pull/140
- fixes https://github.com/ipa320/setup_cob4/issues/149
- improved readability of code
- reduces duplicate code (IntsallUpstart vs UpdateUpstart)
- improuvs usability of Postinstall script for upstart installation

### CODE REVIEW

File: PostInstallCob4.sh [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
-  3.  Setup the robot bashrc and workspace \n
-  4.  Add and setup the mimic user configuration \n
-  5.  Install Upstart software \n
-  6.  Update Upstart software \n
-  7.  Setup the udev rules for the scanners 
+1. Update root ssh keys\n
+2. Synchronize robot user\n
+3. Setup robot bashrc and Workspace\n
+4. Setup mimic user\n
+5. Install upstart\n
```
Comment (ipa-nhg) [inline PostInstallCob4.sh]: for me make more sense separate both functions Install and Update, but should be also ok

File: PostInstallCob4.sh [hunk 2]
```diff
@@ -174,11 +176,12 @@ function SetupMimicUser {
...
   fi
 
-  cob-adduser mimic
+  /u/robot/git/setup_cob4/cob-adduser mimic
+  ssh-copy-id mimic@localhost
 
   GDM_PATH=/etc/gdm/custom.conf
-  sudo ssh $pc_head "sed -i \"s/#  AutomaticLoginEnable=True'/AutomaticLoginEnable=True'/g\" $GDM_PATH"
-  sudo ssh $pc_head "sed -i \"s/#  AutomaticLogin=user1'/AutomaticLogin=mimic'/g\" $GDM_PATH"
+  sudo ssh $pc_head "sed -i \'s/#  AutomaticLoginEnable=True'/AutomaticLoginEnable=True'/g\" $GDM_PATH"
```
[the remark this answers is not in the document]
Comment (ipa-fmw) [reply PostInstallCob4.sh]: good catch. fixed in https://github.com/ipa320/setup_cob4/pull/150/commits/fb13da6326bbe4fe79197556f89365c294c61b8f

File: PostInstallCob4.sh [hunk 3]
```diff
@@ -174,11 +176,12 @@ function SetupMimicUser {
...
+  sudo ssh $pc_head "sed -i \'s/#  AutomaticLogin=user1'/AutomaticLogin=mimic'/g\" $GDM_PATH"
```
[the remark this answers is not in the document]
Comment (ipa-fmw) [reply PostInstallCob4.sh]: good catch. fixed in https://github.com/ipa320/setup_cob4/pull/150/commits/fb13da6326bbe4fe79197556f89365c294c61b8f

File: PostInstallCob4.sh (hunk 1 above)
Comment (ipa-fmw) [reply PostInstallCob4.sh]: calling Install multiple times is possible yo you can simply update with calling Insall upstart again. No need to maintain two functions

**Pull request merged**